2.4/root/usr/share/container-scripts/mongodb/replset_petset_supervisor.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source $CONTAINER_SCRIPTS_PATH/common.sh
+
+info "Waiting for local MongoDB to accept connections ..."
+wait_for_mongo_up &>/dev/null
+
+# We can determine member's number by using its suffix.
+# For example, member with hostname "replica-2" is the _third_ pet.
+MEMBER_ID="${HOSTNAME##*-}"
+MEMBER_HOST="$(hostname -f)"
+
+# Description of possible statuses: https://docs.mongodb.com/manual/reference/replica-states/
+WAIT_PRIMARY_STATUS="while(rs.status().startupStatus || (rs.status().hasOwnProperty(\"myState\") && rs.status().myState != 1)) { printjson(rs.status()); sleep(1000); }; printjson(rs.status());"
+WAIT_SECONDARY_STATUS="var mbrs; while(!mbrs || mbrs.length == 0 || mbrs[0].state != 2) { printjson(rs.status()); sleep(1000); mbrs = rs.status().members.filter(function(el){return el.name.indexOf(\"$MEMBER_HOST\") > -1}); }; print(mbrs[0].stateStr);"
+
+# Initialize replica set only if we're the first member
+if [ "$MEMBER_ID" = '0' ]; then
+    CONFIG="{_id: \"$MONGODB_REPLICA_NAME\", members: [ {_id: 0, host: \"$MEMBER_HOST\"} ]}"
+    info "Initiating MongoDB replica using: $CONFIG"
+    mongo admin --quiet --eval "rs.initiate($CONFIG);$WAIT_PRIMARY_STATUS"
+
+    info "Creating MongoDB users ..."
+    mongo_create_admin
+    mongo_create_user "-u admin -p $MONGODB_ADMIN_PASSWORD"
+
+    info "Successfully initialized replica set"
+else
+    # replica-1.mongodb.myproject.svc.cluster.local -> replica-0.mongodb.myproject.svc.cluster.local
```

Comment:
It shows example of transformation. It helps to skip the code and still understand what I'm trying to do.